driver/src/main/java/org/neo4j/driver/internal/NetworkSession.java

```diff
@@ -139,7 +139,7 @@ public synchronized void reset()
     @Override
     public boolean isOpen()
     {
-        return isOpen.get();
+        return isOpen.get() && connection.isOpen();
```

Comment:
I am not sure if we should add connection.isOpen test here. Thought it is handle, however this method before is just to test if `session.close` is ever called. If you want to change this to `if this session could run more queries` or something else, then make sure that other methods who calling this `isOpen` is not broken.